pkg/analysis/passes/sdkusage/sdkusage.go

```diff
@@ -101,5 +125,82 @@ func run(pass *analysis.Pass) (interface{}, error) {
 		return nil, nil
 	}
 
+	if pluginGoSdkVersion == "" {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			goModError,
+			"go.mod can not be parsed from your source code",
+			"Your go.mod can not be parsed. Please make sure it is valid You can use `go mod tidy` to fix it.",
+		)
+		return nil, nil
+	}
+
+	latestRelease, err := githubapi.FetchLatestGrafanaSdkRelease()
+	if err != nil {
+		// it is most likely this failed because of github auth or rate limits
+		logme.Debugln(err)
+		return nil, nil
+	}
+
+	if latestRelease.TagName == pluginGoSdkVersion {
+		// plugin is using the latest version no further checks
+		return nil, nil
+	}
+
+	// today date in RFC3339 format
+	today := GetNowInRFC3339()
+
+	daysDiff, err := daysDifference(today, latestRelease.PublishedAt)
+	if err != nil {
+		// error calculating the days difference could be a problem in github date format
+		// ignoring it
+		logme.DebugFln("Error calculating days difference: %s", err.Error())
+		return nil, nil
+	}
+
+	if daysDiff > fiveMonths {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			goSdkOlderThanFiveMonths,
+			"Your Grafana go sdk is older than 5 months",
+			"Your Grafana go sdk is older than 5 months. Please upgrade to the latest version",
```

Comment:
Nit: Maybe we should change those messages with guidance on how to upgrade the package? Or at least give the full name of the Go package.

Something like:

```
Your Grafana Go SDK version is older than 5 months. Please upgrade to the latest version by running: "go get -u github.com/grafana/grafana-plugin-sdk-go"
```

Or:

```
Your Go package "github.com/grafana/grafana-plugin-sdk-go" is older than 5 months. Please upgrade to the latest version.
```

I would also change the way it's spelled

`Grafana go sdk` -> `Grafana Go SDK`

WDYT?

Reply:
Good idea. let me change that.